### PR TITLE
[TimerMixin] rm TimerMixin from TimersTest

### PR DESCRIPTION
**Pull request by exced**
Related to #21485

Remove TimerMixin from TimersTest

## Test Plan:
- All flow tests succeed
- `yarn run test-android-instrumentation` && CI integration tests ? (Did not run the command yet and not sure how to properly test it)

## Release Notes:

[GENERAL] [ENHANCEMENT] [TimersTest.js] - rm TimerMixin

### CODE REVIEW

File: IntegrationTests/TimersTest.js [hunk 1]
```diff
@@ -151,12 +164,75 @@ const TimersTest = createReactClass({
...
+      this._timeoutIDs.splice(this._timeoutIDs.indexOf(timeoutID));
+      fn();
+    }, ms);
+    this._timeoutIDs.push(timeoutID);
+    return timeoutID;
+  },
+
+  _setInterval: function(fn: () => void, ms?: number): IntervalID {
+    const intervalID = setInterval(() => {
+      this._intervalIDs.splice(this._intervalIDs.indexOf(intervalID));
```
Comment (RSNara) [inline IntegrationTests/TimersTest.js]: I think this is a bug? When the callback is first called, we'll clear the interval id from `this_intervalIDs`, which means that we'll never clear the interval in `componentWillUnmount`.

**Comment by exced** [comment]
@RSNara You're write I shouldn't clear in _setInterval ! I fixed it

**Comment by RSNara** [comment]
> @RSNara You're write I shouldn't clear in _setInterval ! I fixed it
> I also found out an issue with `splice` method !
> I still don't know if you prefer using `splice(index,1)` or `filter` (what's the most efficient method ?)
> (#21802 has the same issue)

Nice catch on the `Array.prototype.splice`! 

I think your call about using `Array.prototype.filter` is correct. It's less confusing than `Array.prototype.splice`.

**Comment by RSNara** [comment]
@exced, I just noticed that #21748 does the exact same thing. I modified it to take the approach you took in this PR. Since that's already landing internally, I'll close this PR. Thanks for this PR! (And sorry that I didn't notice the conflict earlier 😢).

**Pull request closed without merging**